src/main/migration/1614748794259-RestaurantDeletion.ts

```diff
@@ -0,0 +1,43 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+/**
+ * This migration is to support restaurant cleanup when a promotion is deleted.
+ * * We only want to delete a promotions restaurant if the restaurant is referencing only one more promotion
+ * * Thus create a trigger to execute on for each promotion (row) that is deleted in the promotion table. We can reference
+ * the deleted promotion through `old`. Then create a function to delete the restaurant based on the deleted promotion `old` only if
+ * the restaurant is referencing zero promotions.
+ * * Intro to triggers: https://www.postgresqltutorial.com/creating-first-trigger-postgresql/
+ * */
+export class RestaurantDeletion1614748794259 implements MigrationInterface {
```

Comment:
I looked into this briefly and came across these and was wondering if you think this would accomplish the same thing:

https://github.com/typeorm/typeorm/pull/7105
https://medium.com/platri-it/how-to-delete-orphaned-rows-full-example-with-typeorm-nestjs-and-postgresql-a8baa29c01f8

Seems like a pretty new feature they added, but it looks like it requires a higher version of typeorm

Reply:
Ahhh okay makes sense I got it confused 😅 Seems weird that there isn't a simpler solution cause this seems like it would be a common issue